src/lib/chat/matrix-client.test.ts

```diff
@@ -119,4 +135,87 @@ describe('matrix client', () => {
       expect(getRooms).toHaveBeenCalledOnce();
     });
   });
+
+  describe('getChannels', () => {
+    it('returns only non-direct rooms as channels', async () => {
+      const rooms = [getRoom({ roomId: 'channel-id' }), getRoom({ roomId: 'dm-id' })];
+
+      const getRooms = jest.fn(() => rooms);
+      const getAccountData = getMockAccountData({ id: 'dm-id' });
+
+      const client = subject({
+        createClient: jest.fn(() => getSdkClient({ getRooms, getAccountData })),
+      });
+
+      await client.connect('username', 'token');
+      const channels = await client.getChannels('network-id');
+
+      expect(channels).toHaveLength(1);
+      expect(channels[0].id).toEqual('channel-id');
+    });
+  });
+
+  describe('getConversations', () => {
+    it('returns only direct rooms as conversations', async () => {
+      const rooms = [getRoom({ roomId: 'channel-id' }), getRoom({ roomId: 'dm-id' })];
+
+      const getRooms = jest.fn(() => rooms);
+      const getAccountData = getMockAccountData({ id: 'dm-id' });
+
+      const client = subject({
+        createClient: jest.fn(() => getSdkClient({ getRooms, getAccountData })),
+      });
+
+      await client.connect('username', 'token');
+      const conversations = await client.getConversations();
+
+      expect(conversations).toHaveLength(1);
+      expect(conversations[0].id).toEqual('dm-id');
+    });
+  });
+
+  describe('getAccountData', () => {
+    it('get account data', async () => {
+      const getAccountData = getMockAccountData([{ id: '!abcdefg' }, { id: '!hijklmn' }]);
```

Comment:
`[{ id: '!abcdefg' }, { id: '!hijklmn' }]` doesn't look like the same format we were seeing from the event the other day. Wasn't it more like:
```js
  {
    'some-user-id': ['!abcdefg', '!hijklmn'],
    'other-user-id': [...more-ids...]
  }
```
If so, we should update all the tests to use that format.

Reply:
Yes correct! Updated. Thanks!

```
  const getAccountData = getMockAccountData({
        '@mockuser1': ['!abcdefg'],
        '@mockuser2:': ['!hijklmn', '!opqrstuv'],
      });
```